account_analytic_no_lines/models/account_move_line.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2016 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = 'account.move.line'
+
+    @api.model
+    def _must_create_analytic_line(self, move_line):
+        if not move_line.product_id:
+            return False
+        if not hasattr(move_line.product_id, 'expense_policy'):
+            return False
+        if move_line.product_id.expense_policy == 'no':
+            return False
+        return True
+
+    @api.multi
+    def create_analytic_lines(self):
+        self = self.filtered(self._must_create_analytic_line)
```

Comment:
@ThomasBinsfeld 
The method is never actually called here right?
it should be `self = self.filtered(self._must_create_analytic_line())`
And `_must_create_analytic_line` should have an api.multi decorator and manage multiples account move line. (and remove the `move_line` parameter)

That said, I wonder, why not make it simpler and just totally by pass `create_analytic_lines`?
I am not really confortable with the "expense policy" of the products, but I wonder, it would not make sense for someone who need analytic lines for products with special expense policies to install this module right?

Problem is also present in version 10 and 11.